src/components/Header/index.js

```diff
@@ -0,0 +1,59 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import profileIcon from '../../images/profileIcon.svg';
+import searchIcon from '../../images/searchIcon.svg';
+import './style.css';
+import SearchBar from '../SearchBar';
+
+function Header({ title, match }) {
+  const [search, setSearch] = useState(false);
+
+  const handleSearch = () => (
+    search ? setSearch(false) : setSearch(true)
+  );
+
+  const renderButtonSearch = () => {
+    if (!match) return;
+
+    const { path } = match;
+    if (path === '/bebidas'
+      || path === '/comidas'
+      || path === '/explorar/comidas/area') {
+      return (
+        <button
+          type="button"
+          onClick={ handleSearch }
```

Comment:
```suggestion
          onClick={ () => setSearch(!search) }
```
Sendo alterado o estado no próprio onClick